src/librustc/diagnostics.rs

```diff
@@ -353,26 +353,35 @@ If you really want global mutable state, try using `static mut` or a global
 "##,
 
 E0018: r##"
-The value of static and const variables must be known at compile time. You
-can't cast a pointer as an integer because we can't know what value the
-address will take.
-
-However, pointers to other constants' addresses are allowed in constants,
-example:
+The value of static and constant integers must be known at compile time. You
```

Comment:
Maybe

> The value of statics and constants must be computable at compile time.

? This way we do not restrict ourselves to any particular kind of values and it still works out well for the counter-case presented by @mahkoh in the issue, because in LLVM-slang `&symbol` is simply a `@symbol` (i.e. still computable during compilation).